### PR TITLE
feat(CB2-5707): Amend TRL Purchaser section for a provisional Tech Record

### PR DESCRIPTION
**Pull request by HazelWilderspin**
Amend TRL Purchaser section for a provisional Tech Record

Added: validators max-length, required. Changed: type interface to for optional.
https://dvsa.atlassian.net/browse/CB2-5707

## Checklist

- [x] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [x] Code and UI has been tested manually after the additional changes
- [x] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [x] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Comment by HazelWilderspin** [comment]
Parking PR until backend changes have been merged to accommodate empty fields

**Review by naathanbrown — DISMISSED** [review]

File: src/app/forms/templates/general/approval-type.template.ts [hunk 1]
```diff
@@ -15,9 +15,7 @@ export function getTypeApprovalSection(vehicleType: VehicleTypes): FormNode {
         type: FormNodeTypes.CONTROL,
         editType: FormNodeEditTypes.SELECT,
         options: getOptionsFromEnum(approvalType),
-        validators: [
-          { name: ValidatorNames.Required }
-        ],
+        validators: vehicleType === VehicleTypes.PSV ? [] : [{ name: ValidatorNames.Required }],
```
Comment (aaron-downing) [inline src/app/forms/templates/general/approval-type.template.ts:18]: Believe this is mandatory for all vehicle types https://dvsa.atlassian.net/wiki/spaces/HVT/pages/49679745/Amend+Technical+Record

Comment (naathanbrown) [reply src/app/forms/templates/general/approval-type.template.ts:18]: 👍

**Review by aaron-downing — APPROVED** [review]

File: src/app/forms/templates/general/approval-type.template.ts (hunk 1 above)
Comment (HazelWilderspin) [reply src/app/forms/templates/general/approval-type.template.ts]: fixt

**Review by naathanbrown — APPROVED** [review]

**Pull request merged**